spec/lib/terraform/runner_spec.rb

```diff
@@ -198,19 +220,80 @@ def verify_req(req)
+        expect(body["cloud_providers"]).to(eq([]))
+      end
+
+      before do
+        ENV["TERRAFORM_RUNNER_URL"] = "https://1.2.3.4:7000"
+
+        update_stub = stub_request(:post, "https://1.2.3.4:7000/api/stack/apply")
+                      .with { |req| verify_req(req) }
+                      .to_return(
+                        :status => 200,
+                        :body   => @hello_world_update_response.to_json
+                      )
+
+        retrieve_stub = stub_request(:post, "https://1.2.3.4:7000/api/stack/retrieve")
+                        .with(:body => hash_including({:stack_id => @hello_world_retrieve_update_response['stack_id']}))
+                        .to_return(
+                          :status => 200,
+                          :body   => @hello_world_retrieve_update_response.to_json
+                        )
+      end
+
+      let(:input_vars) { {'name' => 'Future-World'} }
+
+      let(:input_vars_type_constraints) do
+        {
+          "name" => {"name" => "name", "label" => "Name", "type" => "string", "description" => "name is required", "required" => true, "secured" => false, "hidden" => false, "immutable" => false, "default" => "World"},
+        }
+      end
+
+      it ".run update stack for reconfiguration of stack created with hello-world terraform template" do
+        async_response = Terraform::Runner.run(
+          Terraform::Runner::ActionType::UPDATE,
+          File.join(__dir__, "runner/data/hello-world"),
+          {
+            :input_vars                  => input_vars,
+            :input_vars_type_constraints => input_vars_type_constraints,
+            :stack_id                    => @hello_world_retrieve_update_response['stack_id']
+          }
+        )
+        expect(update_stub).to(have_been_requested.times(1))
+
+        response = async_response.response
+        expect(retrieve_stub).to(have_been_requested.times(1))
+        expect(response.stack_id).to(eq(@hello_world_update_response['stack_id']))
+        expect(response.action).to(eq('APPLY'))
+        expect(response.stack_name).to(eq(@hello_world_update_response['stack_name']))
+
+        expect(response.status).to(eq('SUCCESS'), "terraform-runner failed with:\n#{response.status}")
+        expect(response.message).to(include('Apply complete! Resources: 1 added, 0 changed, 1 destroyed.'))
+      end
+    end
+
+    describe 'Delete stack for Retirement of created stack' do
       delete_stub = nil
       delete_retrieve_stub = nil
 
       def verify_req(req)
         body = JSON.parse(req.body)
         expect(body["stack_id"]).to(eq(@hello_world_retrieve_delete_response['stack_id']))
         expect(body).to(have_key('templateZipFile'))
-        expect(body["parameters"]).to(eq([{"name" => "name", "value" => "New World", "secured" => "false"}]))
+        expect(body["parameters"]).to(eq([{"name" => "name", "value" => "Future-World", "secured" => "false"}]))
         expect(body["cloud_providers"]).to(eq([]))
```

Comment:
Similar comment to above, the style we use for rspec expections is not to use parens around the `.to(eq())` so `.to eq(...)` is preferred